File: app/src/main/java/com/ssindher11/todonotes/adapter/NotesAdapter.java
package com.ssindher11.todonotes.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.ssindher11.todonotes.R;
import com.ssindher11.todonotes.clicklisteners.ItemClickListener;
import com.ssindher11.todonotes.model.Notes;

import java.util.List;

public class NotesAdapter extends RecyclerView.Adapter<NotesAdapter.ViewHolder> {

    private List<Notes> notesList;
    private ItemClickListener itemClickListener;

    public NotesAdapter(List<Notes> notesList, ItemClickListener itemClickListener) {
        this.notesList = notesList;
        this.itemClickListener = itemClickListener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.notes_adapter_layout, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        final Notes notes = notesList.get(position);
        final String title = notes.getTitle();
        final String description = notes.getDescription();

        holder.titleTV.setText(title);
        holder.descTV.setText(description);
        holder.itemView.setOnClickListener(view -> itemClickListener.onClick(notes));
    }


    @Override
    public int getItemCount() {
        return notesList.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        TextView titleTV, descTV;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            titleTV = itemView.findViewById(R.id.tv_title);
            descTV = itemView.findViewById(R.id.tv_description);
        }
    }
}

File: app/src/main/java/com/ssindher11/todonotes/adapter/NotesAdapter.kt
package com.ssindher11.todonotes.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.ssindher11.todonotes.R
import com.ssindher11.todonotes.clicklisteners.ItemClickListener
import com.ssindher11.todonotes.db.Notes

class NotesAdapter(val list: List<Notes>, val itemClickListener: ItemClickListener) : RecyclerView.Adapter<NotesAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view: View = LayoutInflater.from(parent.context).inflate(R.layout.notes_adapter_layout, parent, false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int = list.size

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val notes: Notes = list[position]
        val title = notes.title
        val description = notes.description

        holder.titleTV.text = title
        holder.descTV.text = description
        holder.markStatusCB.isChecked = notes.isTaskCompleted
        Glide.with(holder.itemView).load(notes.imagePath).into(holder.previewIV)

        holder.itemView.setOnClickListener { itemClickListener.onClick(notes) }
        holder.markStatusCB.setOnCheckedChangeListener { _, isChecked ->
            notes.isTaskCompleted = isChecked
            itemClickListener.onUpdate(notes)
        }
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val titleTV: TextView = itemView.findViewById(R.id.tv_title)
        val descTV: TextView = itemView.findViewById(R.id.tv_description)
        val markStatusCB: CheckBox = itemView.findViewById(R.id.cb_mark_status)
        val previewIV: ImageView = itemView.findViewById(R.id.iv_image_preview)
    }
}